server/kv.go

```diff
@@ -208,17 +214,36 @@ func (p *Plugin) DeleteJiraInstance(key string) (returnErr error) {
 		if appErr != nil {
 			return appErr
 		}
+		p.updateConfig(func(conf *config) {
+			// Reset, will get re-initialized as needed
+			conf.currentInstance = nil
+			conf.currentInstanceExpires = time.Time{}
+		})
 		p.debugf("Deleted: current Jira instance")
 	}
 
 	return nil
 }
 
 func (p *Plugin) LoadCurrentJIRAInstance() (Instance, error) {
+	conf := p.getConfig()
+	now := time.Now()
+
+	if now.Before(conf.currentInstanceExpires) {
+		if conf.currentInstance == nil {
```

Comment:
Why would this happen? Maybe just check `conf.currentInstance == nil` above and let it try to load it if it's nil?